### ISSUE TITLE
Run the action with GitHub Action's `merge_group` event

### ISSUE THREAD

**Issue by YossiSaadi** [wagoid/commitlint-github-action#757]
## TL;DR:
I'm using the `merge_group` event on GitHub, to check against the squashed commit that is about to be merged, and using merge queue only for this purpose, to prevent merge commits to not be commitlint.
For some reason, the action keeps on succeeding although it should fail.
----

### An important note:
When I run this action in the `pull_request` event or `push` event it works correctly and fails the workflow!
so there's only an issue with the `merge_group` event.

**See this repo:** https://github.com/yossi-test-org/test-merge-queue
**with this action:** https://github.com/yossi-test-org/test-merge-queue/actions/workflows/commitlint.yml
**and more specifically this action run:** https://github.com/yossi-test-org/test-merge-queue/actions/runs/7897818307/job/21554034430
**which ran on this commit:** https://github.com/yossi-test-org/test-merge-queue/commit/2f49ddbde246e380780396787ae33e9842e198e5

Attached an image to summarize all (see the "Show commit message" step that makes sure what commitlint runs against):
![image](https://github.com/wagoid/commitlint-github-action/assets/30905362/603625a0-eec7-4de6-a16c-e95e1611ef1a)

```yml
name: Lint merge commit name
on:
  pull_request:
  merge_group:
    types: [checks_requested]
jobs:
  lint-commit-name:
    if: ${{ github.event_name == 'merge_group' }}
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 1
      - uses: wagoid/commitlint-github-action@v5
```

It's a required action, so it should fail the merge queue and to not merge this PR:
![image](https://github.com/wagoid/commitlint-github-action/assets/30905362/bf818588-04e3-4262-93c7-48ab2543f49c)
![image](https://github.com/wagoid/commitlint-github-action/assets/30905362/2a47bed1-f4aa-46f6-9a9d-5d4dc74fba22)

**Comment by YossiSaadi** [comment]
I've forked created a PR #806 
also tested it and it seem to work perfect :) 

The PR:
https://github.com/yossi-test-org/test-merge-queue/pull/24
![image](https://github.com/user-attachments/assets/084908da-c399-40c6-8c5e-a2c13c2c44b1)

with bad squashed commit:
https://github.com/yossi-test-org/test-merge-queue/actions/runs/12221132350/job/34089796604
![image](https://github.com/user-attachments/assets/4d9a96cf-f6e9-4a34-90f1-3bed7da9027d)

with good squashed commit:
https://github.com/yossi-test-org/test-merge-queue/actions/runs/12221143821/job/34089822554
![image](https://github.com/user-attachments/assets/c3598850-ce29-41c3-88fe-ceb2a9c78bb2)

**Comment by wagoid** [comment]
Thanks for opening the [PR](https://github.com/wagoid/commitlint-github-action/pull/806) and fixing this! I'm closing the issue now
